x/incentives/client/cli/query.go

```diff
@@ -384,31 +386,77 @@ $ %s query incentives rewards-estimation
 				return err
 			}
 
-			lockIdStrs := strings.Split(lockIdsCombined, ",")
-			lockIds := []uint64{}
-			for _, lockIdStr := range lockIdStrs {
-				lockId, err := strconv.ParseUint(lockIdStr, 10, 64)
+			// GetCmdAccountLockedLongerDuration
+			var res *lockuptypes.AccountLockedLongerDurationResponse
+			if owner != "" {
+				queryClientLockup := lockuptypes.NewQueryClient(clientCtx)
+
+				res, err = queryClientLockup.AccountLockedLongerDuration(cmd.Context(), &lockuptypes.AccountLockedLongerDurationRequest{Owner: owner, Duration: time.Second})
 				if err != nil {
 					return err
 				}
-				lockIds = append(lockIds, lockId)
+			} else {
+				owner = ""
+				//res = nil
+			}
+
+			ownerLocks := []uint64{}
+
+			if res != nil {
+				for _, lockId := range res.Locks {
+					ownerLocks = append(ownerLocks, lockId.ID)
+				}
+			}
+
+			lockIdStrs := strings.Split(lockIdsCombined, ",")
+			lockIds := []uint64{}
+			if lockIdsCombined == "" && owner == "" {
+				err = fmt.Errorf("If owner flag is not set, lock IDs must be provided")
+				return err
+			} else if lockIdsCombined != "" {
+				for _, lockIdStr := range lockIdStrs {
+					lockId, err := strconv.ParseUint(lockIdStr, 10, 64)
+					if err != nil {
+						return err
+					}
+					lockIds = append(lockIds, lockId)
+				}
+			} else if lockIdsCombined == "" && owner != "" {
+				for _, lockId := range ownerLocks {
+					lockIds = append(lockIds, lockId)
+				}
+			}
+
+			if len(lockIds) != 0 && owner != "" {
+				for _, lockId := range lockIds {
+					validInputLockId := contains(ownerLocks, lockId)
+					if !validInputLockId {
+						return fmt.Errorf("Lock-id %v does not belong to %v", lockId, owner)
+					}
+				}
 			}
 
 			endEpoch, err := cmd.Flags().GetInt64(FlagEndEpoch)
 			if err != nil {
 				return err
 			}
 
-			res, err := queryClient.RewardsEst(cmd.Context(), &types.RewardsEstRequest{
+			// TODO: Figure out why some lock ids are good and some causes "Error: rpc error: code = Unknown desc = panic message redacted to hide potentially sensitive system info: panic"
```

Comment:
TODO: Figure out why some lock ids are good and some causes "Error: rpc error: code = Unknown desc = panic message redacted to hide potentially sensitive system info: panic"

Reply:
My guess would be that lock ID was probably deleted from mainnet. (Finished unlocking)

Reply:
@ValarDragon  So for some addresses its fine but lets use this random one I found on mintscan as an example:

`osmosisd q incentives rewards-estimation --owner osmo1lqxwdzlxvvh700sxagcm5flw2tpmp890s76lsj --end-epoch 400`

Using this branch will give you a panic (using main wont work at all since this query has been broken)

You can use `osmosisd q lockup account-locked-longer-duration osmo1lqxwdzlxvvh700sxagcm5flw2tpmp890s76lsj 1s` to see all the lock IDs and then query the rewards estimation to see which locks cause this panic, but I have yet to see a clear pattern to which locks cause panics and which dont

Reply:
This address though, with multiple locks, gives a valid response:

`osmosisd q incentives rewards-estimation --owner osmo13gx9q6q8cxhym9pzyvjtkgagjqrgwxe0gcleyh --end-epoch 400`